packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js

```diff
@@ -601,5 +601,131 @@ describe('plugin-conversation', function () {
+        }).then(({reaction}) => {
+          assert.include(reaction, parent.id);
+        }));
+    });
+
+    describe('#listChildActivitiesByParentId()', () => {
+      let conversation, parent;
+      let replies;
+
+      before('create conversation with thread replies', () => webex.internal.conversation.create({participants})
+        .then((c) => {
+          conversation = c;
+          assert.lengthOf(conversation.participants.items, 3);
+
+          return webex.internal.conversation.post(conversation, {displayName: 'first message'});
+        }).then((parentAct) => {
+          parent = parentAct;
+
+          const messages = ['thread 1', 'thread 2', 'thread 3'];
+
+          return Promise.all(messages.map((msg) =>
+            webex.internal.conversation.post(conversation, msg, {
+              parentActivityId: parent.id,
+              activityType: 'reply'
+            })));
+        }).then((repliesArr) => {
+          replies = repliesArr;
+        }));
+
+      it('retrieves thread reply activities for a given parent', () => webex.internal.conversation.listChildActivitiesByParentId(conversation.url, parent.id, 'reply')
+        .then((response) => {
+          const {items} = response.body;
+
+          items.forEach((threadAct) => {
+            assert.include(replies.map((reply) => reply.id), threadAct.id);
+          });
+        }));
+    });
+
+    describe('#getReactionSummaryByParentId()', () => {
+      let conversation, parent;
+      const reactionType = 'celebrate';
+
+      before('create conversation with an activity and reactions', () => webex.internal.conversation.create({participants})
+        .then((c) => {
+          conversation = c;
+          assert.lengthOf(conversation.participants.items, 3);
+
+          return webex.internal.conversation.post(conversation, {displayName: 'first message'});
+        })
+        .then((parentAct) => {
+          parent = parentAct;
+
+          return webex.internal.conversation.addReaction(conversation, reactionType, parent);
+        }));
+
+      const reactionFetch = () => webex.internal.conversation.getReactionSummaryByParentId(conversation.url, parent.id, {activityType: 'reactionSummary', includeChildren: true});
+      const testReactionResponse = (reactionSummaries) => {
+        assert.equal(reactionSummaries.length, 2);
+        reactionSummaries.forEach((summary) => {
+          assert.equal(summary.activity.object.reactions[0].displayName, reactionType);
+        });
+      };
+      const handleReactionTest = (doneCb) => {
+        reactionFetch().then((summary) => {
+          if (summary.length) {
+            testReactionResponse(summary);
+            doneCb();
+          }
+          else {
+            setTimeout(() => handleReactionTest(doneCb), 5000);
+          }
+        });
+      };
+
+      it('retrieves reactionSummarys and reactionSelfSummarys', (done) => {
+        handleReactionTest(done);
```

Comment:
Not a big fan of this type of test format. If there's a way to add the assertions and actions within the `it` block, that would be preferred